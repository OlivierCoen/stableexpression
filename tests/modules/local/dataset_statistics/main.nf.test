nextflow_process {

    name "Test Process DATASET_STATISTICS"
    script "modules/local/dataset_statistics/main.nf"
    process "DATASET_STATISTICS"
    tag "dataset_statistics"
    tag "module"

    test("Should run without failures") {

        when {
            process {
                """
                meta = [dataset: 'test']
                count_file = file( '$projectDir/tests/test_data/dataset_statistics/input/count.raw.cpm.quant_norm.parquet', checkIfExists: true)
                input[0] = [meta, count_file]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
