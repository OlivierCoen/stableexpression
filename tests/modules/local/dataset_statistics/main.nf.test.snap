{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "dataset": "test"
                        },
                        "count.raw.cpm.dataset_stats.csv:md5,93290e92d7ac9c0abdd2d9e76a7aff6b"
                    ]
                ],
                "1": [
                    [
                        "DATASET_STATISTICS",
                        "python",
                        "3.12.8"
                    ]
                ],
                "2": [
                    [
                        "DATASET_STATISTICS",
                        "pandas",
                        "2.2.3"
                    ]
                ],
                "3": [
                    [
                        "DATASET_STATISTICS",
                        "scipy",
                        "1.15.0"
                    ]
                ],
                "4": [
                    [
                        "DATASET_STATISTICS",
                        "pyarrow",
                        "19.0.0"
                    ]
                ],
                "stats": [
                    [
                        {
                            "dataset": "test"
                        },
                        "count.raw.cpm.dataset_stats.csv:md5,93290e92d7ac9c0abdd2d9e76a7aff6b"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-08T19:39:30.385303512"
    }
}